### PR TITLE
JESD204 FSM improvements

### PR DESCRIPTION
**Pull request by dbogdan**
## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [x] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: util/no_os_clk.c [hunk 1]
```diff
@@ -79,13 +89,20 @@ int32_t no_os_clk_init(struct no_os_clk_desc **desc,
...
-	if (!desc->platform_ops->remove)
-		return -ENOSYS;
+	if (desc->platform_ops->remove) {
+		ret = desc->platform_ops->remove(desc);
+		if (ret)
+			return ret;
+	}
 
-	return desc->platform_ops->remove(desc);
+	free(desc);
```
Comment (danmois) [inline util/no_os_clk.c:103]: Use no_os_free().

File: projects/ad9081/src/app_jesd.c [hunk 2]
```diff
@@ -162,27 +164,17 @@ int32_t app_jesd_init(struct no_os_clk clk[2],
...
-	tx_jesd_init.lane_clk.hw_ch_num = 0;
-	tx_jesd_init.lane_clk.name = "jesd_tx";
-	tx_jesd_init.lane_clk.platform_ops = &jesd204_clk_ops;
+	tx_jesd_init.lane_clk = tx_adxcvr->clk_out;
 
 	ret = axi_jesd204_tx_init_jesd_fsm(&tx_jesd, &tx_jesd_init);
 	if (ret)
 		return ret;
 
-	clk[1].clk_desc = tx_jesd->lane_clk->clk_desc;
```
Comment (danmois) [inline projects/ad9081/src/app_jesd.c:179]: Required by ad9081_init_param.

File: projects/ad9081/src/app_jesd.c [hunk 3]
```diff
@@ -162,27 +164,17 @@ int32_t app_jesd_init(struct no_os_clk clk[2],
...
-
 	ret = axi_jesd204_rx_init_jesd_fsm(&rx_jesd, &rx_jesd_init);
 	if (ret)
 		return ret;
 
-	clk[0].clk_desc = tx_jesd->lane_clk->clk_desc;
```
Comment (danmois) [inline projects/ad9081/src/app_jesd.c:185]: Required by ad9081_init_param.

**Comment by dbogdan** [comment]
v2:
 - no_os_free()
 - revert ad9081 app_jesd clk[] removal
 - astyle fixes

File: projects/ad9081/src/app_jesd.c (hunk 2 above)
Comment (dbogdan) [reply projects/ad9081/src/app_jesd.c:179]: Fixed.

File: projects/ad9081/src/app_jesd.c (hunk 3 above)
Comment (dbogdan) [reply projects/ad9081/src/app_jesd.c:185]: Fixed.

File: util/no_os_clk.c (hunk 1 above)
Comment (dbogdan) [reply util/no_os_clk.c]: Done.

**Comment by dbogdan** [comment]
v3:
 - more astyle fixes

**Comment by dbogdan** [comment]
v4:
 - fix "util/no_os_clk.c:77:4: error: Resource leak: clk [resourceLeak]"

**Review by danmois — APPROVED** [review]

**Pull request merged**